src/components/AllocationsComponent.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { Header, Table, Pagination, Container } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import loadAllocationsAction from '../_actions/allocations.actions';
+import SideMenuComponent from '../_components/SideMenuComponent';
+import TableRowComponent from './TableRowComponent';
+import LoaderComponent from './LoaderComponent';
+import '../_css/AllocationsComponent.css';
+
+export class AllocationsComponent extends Component {
+  state = {
+    currentAllocations: [],
+    offset: 0,
+    activePage: 1,
+    limit: 10,
+  }
+
+  componentDidMount() {
+    this.props.loadAllocationsAction();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.allAllocations.length !== prevProps.allAllocations.length) {
+      this.setTableContent();
+    }
+  }
+
+  getTotalPages = () => Math.ceil(this.props.allAllocations.length / this.state.limit);
+
+  setTableContent = () => {
+    const currentAllocations = this.props.allAllocations.slice(this.state.offset,
+      (this.state.activePage * this.state.limit));
+    // format date
+    const dateOptions = {
+      weekday: 'long',
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    }
+    currentAllocations.map(el => {
+      return el.created_at = new Date(el.created_at).toLocaleDateString('en-US', dateOptions);
```

Comment:
There is no point for the assignment

Reply:
It doesn't format the date properly if I don't do the assignment. Or is there something I am missing? @theterminalguy